### DISCUSSION TITLE
Are there any fast method to change metadata for plenty objects in minio storage？

### DISCUSSION THREAD

**Discussion by forestbat** [minio/minio#19176] [Q&A] [answered]
We have serveral buckets which have about 10M objects need to be add metadata，but when I search on the Internet，search engine told me that I must move object out of minio storage by `get_object()` and change its metadata，then upload it back to minio server.
This method will cost ~0.3s for each I/O，and 3 million seconds' cost is too long for us.
So are there any faster method to achieve our goal？Or will you add API for change metadata in storage directly in future？

**Comment by harshavardhana** [comment, marked as answer]
Please subscribe for performance review of your cluster https://min.io/pricing
